Sources/Kafka/Configuration/KafkaConfiguration+Metrics.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-client open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-client project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-client project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Metrics
+
+extension KafkaConfiguration {
+    // MARK: - Metrics
+
+    /// Configuration for the metrics emitted by `SwiftKafka`.
+    public struct Metrics: Sendable {
```

Comment:
I think this looks really good now. The only thing left is that I think we should split this into two structs on inside the producer and one inside the consumer configuration and only pull out the fields that make sense to them. Then we should be good to merge this.